mmdet/core/anchor/anchor_generator.py

```diff
@@ -74,8 +77,9 @@ def __init__(self,
                 f'{strides} and {centers}'
 
         # calculate base sizes of anchors
-        self.strides = strides
-        self.base_sizes = list(strides) if base_sizes is None else base_sizes
+        self.strides = [_pair(stride) for stride in strides]
+        self.base_sizes = [min(stride) for stride in strides
```

Comment:
We might need a new unit test to test the new behavior.

Reply:
@ZwwWayne , I added a new test that verifies that the new behavior is equivalent to the old when passing square strides.

If you want another test, please let me know what kind of test.

Reply:
I think the test case when the strides are different might be also necessary since that is what this refactoring for.

Reply:
You're right. I added one that tests the behavior of the strides.